solution/index.js

```diff
@@ -0,0 +1,337 @@
+                li.textContent = tod;
+                li.classList.add("task");
+                li.setAttribute("draggable","true");
+                li.setAttribute("ondragstart","drag(event)");
+                li.setAttribute("id",Math.random());
+                i.append(li);
+            }   
+        }
+        if(i.classList.contains("in-progress-tasks")){
+            for(let todo of JSON.parse(localStorage.getItem("tasks"))["in-progress"]){
+                const li= document.createElement("li");
+                li.textContent = todo;
+                li.classList.add("task");
+                li.setAttribute("draggable","true");
+                li.setAttribute("ondragstart","drag(event)");
+                li.setAttribute("id",Math.random());
+                i.append(li);
+            }  
+        }
+        if(i.classList.contains("done-tasks")) {
+            for(let todo of JSON.parse(localStorage.getItem("tasks"))["done"]) {
+                const li= document.createElement("li");
+                li.textContent = todo;
+                li.classList.add("task");
+                li.setAttribute("draggable","true");
+                li.setAttribute("ondragstart","drag(event)");
+                li.setAttribute("id",Math.random());
+                i.append(li);
+            }    
+        }
+    }
+}
+
+
+
+
+document.addEventListener("dblclick", editTask)//adding event listner to edit task
+function editTask (event)
+{
+    if (event.target.classList.contains("task")){//target to only tasks
+        const lis = event.target;
+        const oldValue = lis.textContent;
+        lis.setAttribute("contenteditable",true)
+        lis.focus(); 
+        lis.addEventListener("blur", Blur);
+        function Blur (){
+            const newValue = lis.textContent;
+            const section = lis.closest("section");
+            const buttonId = section.querySelector("button").id;
+            let arrName = section.id;
+            
+            if(lis.textContent === ""){//if changing to an empty task
+                alert("you must enter value"); // alert
+                lis.remove();
+                let index = data[arrName].indexOf(oldValue);
+                data[arrName].splice(index,1);}
+            else{
+                blurTask (buttonId,arrName,oldValue,newValue)
+            }
+            localStorage.setItem("tasks",JSON.stringify(data))
+
+        }
+    }
+
+}
+
+function blurTask (buttonId,arrName,oldValue,newValue){
+
+    if ((buttonId === "submit-add-to-do")||
+        (buttonId === "submit-add-in-progress")||
+        (buttonId === "submit-add-done")){
+
+            let index = data[arrName].indexOf(oldValue)
+            data[arrName].splice(index,1,newValue)
+    }
+
+}
+
+
+
+///changing task place by alt+(1,2,3)
+document.querySelectorAll('.task').forEach(item => {
+    item.addEventListener('mouseover', (e)=>switchTasks(e))  })
+
+    let specificLis;
+function switchTasks(e)
+{
+    specificLis=e.target;
```

Comment:
That is not a good variable name.
But it doesn't matter, because you shouldn't use global variables anyway. Find another way to achieve what you want.